### PR TITLE
Produce deprecation warnings for deprecated names, remove unused imports

### PR DESCRIPTION
**Pull request by jamadden**
In interfaces.py and registry.py and hookable.py

See https://github.com/zopefoundation/zope.component/pull/36#discussion_r149631796

Do this using zope.deferredimport and zope.deprecation, two new dependencies. This introduces a transitive dependency on zope.proxy, but that was already part of the 'security' extra. zope.proxy runs on pypy but it doesn't yet support making the C extension optional (zopefoundation/zope.proxy#26)

Also drop the use of _compat._BLANK everywhere and just use the literal.

Also remove unused but not BBB exported imports from _api.py in its own commit.

### CODE REVIEW

File: src/zope/component/_api.py [hunk 1]
```diff
@@ -14,22 +14,13 @@
...
-from zope.component.interfaces import IComponentRegistrationConvenience
 from zope.component.interfaces import IFactory
-from zope.component.interfaces import ComponentLookupError
-from zope.component.interfaces import IComponentLookup
-from zope.component._compat import _BLANK
-from zope.component._declaration import adaptedBy
-from zope.component._declaration import adapter
-from zope.component._declaration import adapts
+from zope.interface.interfaces import ComponentLookupError
+from zope.interface.interfaces import IComponentLookup
```
Comment (mgedmin) [inline src/zope/component/_api.py]: Import ordering: stdlib (sys), 3rd-party (zope.hookable, zope.interface), then zope.component.

Comment (jamadden) [reply src/zope/component/_api.py]: That's twice in two days that I've been bitten by import groups in reviews. I'm beginning to appreciate Plone's policy of [just ignoring that](https://docs.plone.org/develop/styleguide/python.html#grouping-and-sorting) 😉

**Comment by jamadden** [comment]
Thanks!

**Comment by icemac** [comment]
@jamadden Maybe we could even work towards a 5.0 release which no longer contains these deprecation warnings as at least some of them are about 6 years old now.

**Comment by jamadden** [comment]
@icemac Sure, we could absolutely eventually do that. I'm in no particular hurry though 😄 Now that the deprecation warnings are in place, the maintenance burden of them is pretty small (though I will admit there is a cognitive burden to those reading the code for the first time, a bit of technical debt).

**Pull request merged**